salon_creator/lib/screens/salon_creation_screen.dart

```diff
@@ -8,20 +20,529 @@ class SalonCreationScreen extends StatefulWidget {
+              ),
+            );
+            Navigator.of(context).pushReplacementNamed('/home');
+          },
+          child: Text("スキップ"),
+        ),
+      ],
+      title: Text("サロン作成"),
+    );
+  }
+
+  Widget _buildSalonPriceForm() {
+    return Row(
+      crossAxisAlignment: CrossAxisAlignment.end,
+      children: [
+        TextFieldWithLabel(
+          onChanged: (_) => _updateScreenContext(),
+          controller: priceController,
+          title: "月額料金",
+          hintText: "1000",
+          width: 100,
+          keyboardType: TextInputType.number,
+        ),
+        SizedBox(
+          width: 8,
+        ),
+        Text(
+          "¥",
+          style: TextStyle(fontSize: 24),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildContentForm() {
+    return Column(
+      children: [
+        CustomLabel(
+          title: "説明文",
+        ),
+        SizedBox(
+          height: 12,
+        ),
+        _buildDescriptionTextField(),
+      ],
+    );
+  }
+
+  Widget _buildDescriptionTextField() {
+    return Container(
+      height: 216,
+      child: TextField(
+        onChanged: (_) => _updateScreenContext(),
+        controller: descriptionController,
+        textAlignVertical: TextAlignVertical.center,
+        keyboardType: TextInputType.multiline,
+        maxLength: 300,
+        maxLines: 100,
+        style: TextStyle(
+          fontSize: 16,
+        ),
+        decoration: InputDecoration(
+          contentPadding: EdgeInsets.fromLTRB(16, 8, 16, 8),
+          focusedBorder: OutlineInputBorder(
+            borderSide: BorderSide(
+              color: Colors.black,
+              width: 0.5,
+            ),
+          ),
+          enabledBorder: OutlineInputBorder(
+            borderSide: BorderSide(
+              color: Colors.black,
+              width: 0.5,
+            ),
+          ),
+          hintText: "サロンの目的、発信内容等",
+        ),
+      ),
+    );
+  }
+
+  Future saveSalonDetail() async {
+    final dbHandle = DbHandler();
+    Salon salon = Salon();
+    salon.category = categoriesController.text;
+    salon.description = descriptionController.text;
+    salon.name = salonNameController.text;
+    salon.owner = FirebaseAuth.instance.currentUser.email;
+    salon.price = priceController.text;
+    if (_imageFiles.isNotEmpty) {
+      salon.media = await _uploadImageAndGetUrl();
+    }
+    await dbHandle.addSalon(salon);
+  }
+
+  Future<void> _storeMediaFiles(BuildContext context) async {
+    try {
+      Navigator.of(context).pop();
+      final pickedFile =
+          await imagePicker.pickImage(source: ImageSource.gallery);
+
+      setState(() {
+        _imageFiles.add(pickedFile);
+        index = _imageFiles.length - 1;
+      });
+      WidgetsBinding.instance.addPostFrameCallback((_) => _scrollToRight());
+      _updateScreenContext();
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  Future _uploadImageAndGetUrl() async {
```

Comment:
I think this function is being used elsewhere as well. Why not move it to a class named StorageHandler (similar to DbHandler) and keep it there for common use?